File: stack_queue/주식가격.py
def solution(prices):
    n = len(prices)
    answer = [0 for i in range(len(prices))]

    for i in range(n):
        for j in range(i + 1, n):
            if prices[i] > prices[j]:
                answer[i] = j - i
                break
        if answer[i] == 0:
            answer[i] = n - 1 - i

    return answer

File: src/stackQueue/주식가격.java
package stackQueue;

import java.util.*;

public class 주식가격 {
    public int[] solution(int[] prices) {
        Stack<Integer> priceRaise = new Stack<>();
        int[] answer = new int[prices.length];

        for(int i = 0 ; i < prices.length ; i++){
            while(!priceRaise.isEmpty() && prices[i] < prices[priceRaise.peek()]){
                int pushTime = priceRaise.pop();
                answer[pushTime] = i - pushTime;
            }
            priceRaise.push(i);
        }

        while (!priceRaise.isEmpty()) {
            int pushTime = priceRaise.pop();
            answer[pushTime] = prices.length - pushTime - 1;
        }

        return answer;
    }
}
